zingo-testutils/src/lib.rs

```diff
@@ -737,6 +738,131 @@ pub mod scenarios {
+                &get_base_address!(recipient, "unified"),
+                value.checked_div(2).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // received from a faucet to sapling
+        faucet
+            .do_send(vec![(
+                &get_base_address!(recipient, "sapling"),
+                value.checked_div(4).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // received from a faucet to transparent
+        faucet
+            .do_send(vec![(
+                &get_base_address!(recipient, "transparent"),
+                value.checked_div(4).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // send to a faucet
+        recipient
+            .do_send(vec![(
+                &get_base_address!(faucet, "unified"),
+                value.checked_div(10).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // send to self orchard
+        recipient
+            .do_send(vec![(
+                &get_base_address!(recipient, "unified"),
+                value.checked_div(10).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // send to self sapling
+        recipient
+            .do_send(vec![(
+                &get_base_address!(recipient, "sapling"),
+                value.checked_div(10).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // send to self transparent
+        recipient
+            .do_send(vec![(
+                &get_base_address!(recipient, "transparent"),
+                value.checked_div(10).unwrap(),
+                None,
+            )])
+            .await
+            .unwrap();
+        increase_height_and_wait_for_client(&scenario_builder.regtest_manager, &recipient, 1)
+            .await
+            .unwrap();
+        // shield transparent
+        recipient
+            .do_shield(&[Pool::Transparent], None)
+            .await
+            .unwrap();
```

Comment:
would it be good to include some of these actions in the kotlin test itself?
so then we can test that android is shielding correctly?
maybe you intend this scenario to just test balance on its own and other tests will test shielding... its up to you